includes/classes/SiteAutomation/Integration.php

```diff
@@ -52,31 +68,49 @@ public function get_curated_posts( $posts, $query ) {
 		}
 
 		$curated_response = $this->request->get( $query_vars['sophi_curated_page'], $query_vars['sophi_curated_widget'] );
+		$request_status   = $this->request->get_status();
 
-		// Determine how we should format the results  based on the fields parameter.
-		$fields = $query->get( 'fields', '' );
+		if ( $request_status['success'] ) {
+			$query->sophi_curated_post_list_success = true;
+			$query->num_posts = count( $curated_response );
 
-		switch ( $fields ) {
-			case 'ids':
-				$new_posts = $this->format_hits_as_ids( $curated_response );
-				break;
+			// Determine how we should format the results  based on the fields parameter.
+			$fields = $query->get( 'fields', '' );
 
-			case 'id=>parent':
-				$new_posts = $this->format_hits_as_id_parents( $curated_response );
-				break;
+			switch ( $fields ) {
+				case 'ids':
+					$new_posts = $this->format_hits_as_ids( $curated_response );
+					break;
 
-			default:
-				$new_posts = $this->format_hits_as_posts( $curated_response );
-				break;
-		}
+				case 'id=>parent':
+					$new_posts = $this->format_hits_as_id_parents( $curated_response );
+					break;
 
-		$new_posts = array_filter( $new_posts );
+				default:
+					$new_posts = $this->format_hits_as_posts( $curated_response );
+					break;
+			}
+		}
 
 		if ( ! empty( $new_posts ) ) {
-			return $new_posts;
+			$posts = array_filter( $new_posts );
 		}
 
-		return $posts;
+		/**
+		 * The curated post list result that is injected to WP_Query.
+		 *
+		 * @since 1.0.9
+		 * @hook sophi_curated_post_list
+		 *
+		 * @param {array} $posts Post list.
+		 * @param {string} $query_vars['sophi_curated_page'] Sophi curated page param.
+		 * @param {string} $query_vars['sophi_curated_widget'] Sophi curated widget param.
+		 * @param {array}  $request_status The request status, whether it was successful or not.
+		 * @param {object} $query Original query.
```

Comment:
Not a blocker @oscarssanchez but as we know the type of object here, it would be good to explicitly state `@param {WP_Query}`.